Modules/_collectionsmodule.c

```diff
@@ -1155,13 +1202,24 @@ deque_contains(dequeobject *deque, PyObject *v)
     return 0;
 }
 
+static int
+deque_contains(dequeobject *deque, PyObject *v)
+{
+    int result;
+    Py_BEGIN_CRITICAL_SECTION(deque);
+    result = deque_contains_lock_held(deque, v);
+    Py_END_CRITICAL_SECTION();
+    return result;
+}
+
 static Py_ssize_t
 deque_len(dequeobject *deque)
 {
-    return Py_SIZE(deque);
+    return _Py_atomic_load_ssize(&((PyVarObject *)deque)->ob_size);
```

Comment:
I thought `Py_SIZE` was nogil-safe? What did I miss? :)

Reply:
We discussed making `Py_SIZE` use an atomic load, but then decided not to do it, at least for now.

`Py_SIZE()` is used frequently in cases where we already hold the lock on the object. Using atomic loads/stores where plain loads/stores are sufficient tends to hide data races from thread sanitizer. It's not a huge deal -- if this leads to a lot of otherwise unnecessary code change, then we can revisit it in the future.

Reply:
Thanks!